crates/generate/src/types/flags.rs

```diff
@@ -41,6 +41,19 @@ pub(super) fn define_flags(names: &Names, name: &witx::Id, f: &witx::FlagsDataty
             pub fn contains(&self, other: &#ident) -> bool {
                 !*self & *other == Self::EMPTY_FLAGS
             }
+
+            // Reading and validation are nearly the same thing for flags, so we define one private
+            // helper method that we use for GuestValue::read and GuestValue::validate
+            fn validate_read(ptr: &wiggle_runtime::GuestPtr<#ident>) -> Result<(*mut #ident, #ident), wiggle_runtime::GuestError> {
+                use std::convert::TryFrom;
+                use wiggle_runtime::GuestType;
+                let host_ptr =
+                    ptr.mem()
+                        .validate_size_align(ptr.offset(), Self::guest_align(), Self::guest_size())?;
+                let reprval = #repr::read(&ptr.cast())?;
+                let value = #ident::try_from(reprval)?;
+                Ok((host_ptr as *mut #ident, value))
+            }
```

Comment:
This is more of a memo for the future: this `validate_read` and the one for enums look exactly the same so there exists a potential for some refactoring here.

Reply:
Yes they’re exactly the same for flags and enums I believe. I throughly about DRY but it didn’t seem like enough benefit to offset readability.

Reply:
Makes sense! On the other hand, in the future, we might still considering refactoring it out should we ever wanna change the logic inside it. Having it multiplied in a couple of places makes it easier to accidentally update both (a rather painful point in favour: snapshot0 in `wasi-common`).